contracts/DMMFactory.sol

```diff
@@ -0,0 +1,122 @@
+pragma solidity 0.6.6;
+
+import "@openzeppelin/contracts/utils/EnumerableSet.sol";
+
+import "./interfaces/IDMMFactory.sol";
+import "./DMMPool.sol";
+
+contract DMMFactory is IDMMFactory {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    uint256 internal constant BPS = 10000;
+
+    address private feeTo;
+    uint16 private governmentFeeBps;
+    address public override feeToSetter;
+
+    mapping(IERC20 => mapping(IERC20 => EnumerableSet.AddressSet)) internal tokenPairs;
+    mapping(IERC20 => mapping(IERC20 => address)) public override getNonAmpPair;
+    address[] public override allPairs;
+
+    event PairCreated(
+        IERC20 indexed token0,
+        IERC20 indexed token1,
+        address pair,
+        uint32 ampBps,
+        uint256 totalPair
+    );
+    event SetFeeConfiguration(address feeTo, uint16 governmentFeeBps);
+    event SetFeeToSetter(address feeToSetter);
+
+    constructor(address _feeToSetter) public {
+        feeToSetter = _feeToSetter;
+    }
+
+    function createPair(
+        IERC20 tokenA,
+        IERC20 tokenB,
+        uint32 ampBps
+    ) external override returns (address pair) {
+        require(tokenA != tokenB, "DMM: IDENTICAL_ADDRESSES");
+        (IERC20 token0, IERC20 token1) = tokenA < tokenB ? (tokenA, tokenB) : (tokenB, tokenA);
+        require(address(token0) != address(0), "DMM: ZERO_ADDRESS");
+        require(ampBps >= BPS, "DMM: INVALID_BPS");
+        // only exist 1 non-amplification pool of a pair.
```

Comment:
// only 1 unamplified (or non-amplified) pool can exist

Reply:
unamplified because that was what was used in the whitepaper

Reply:
change in #29 nonAmp -> unamplified in every places